components/renku_data_services/storage/rclone_patches.py

```diff
@@ -0,0 +1,264 @@
+"""Patches to apply to phe rclone storage schema."""
+
+from copy import deepcopy
+from typing import Any
+
+from renku_data_services import errors
+
+BANNED_STORAGE = {
+    "alias",
+    "crypt",
+    "cache",
+    "chunker",
+    "combine",
+    "compress",
+    "hasher",
+    "local",
+    "memory",
+    "union",
+}
+
+OAUTH_PROVIDERS = [
+    "acd",
+    "box",
+    "drive",
+    "dropbox",
+    "gcs",
+    "gphotos",
+    "hidrive",
+    "jottacloud",
+    "mailru",
+    "onedrive",
+    "pcloud",
+    "pikpak",
+    "premiumzeme",
+    "putio",
+    "sharefile",
+    "yandex",
+    "zoho",
+]
```

Comment:
suggestion: It would be nice if these are both sets probably. Also it does not hurt to use `typing.Final` on them to make sure no one modifies them in some function somewhere else.